full-stack-tests/presentation/src/learning-snippets/Content.test.ts

```diff
@@ -11,582 +11,1002 @@ import { getFieldByLabel, tryGetFieldByLabel } from "../Utils";
+          { label: "Eye Point" },
+          { label: "Focus Distance" },
+          { label: "Is Camera On" },
+          { label: "Is Private" },
+          { label: "Lens Angle" },
+          { label: "Model" },
+          { label: "Model Selector" },
+          { label: "Origin" },
+          { label: "Pitch" },
+          { label: "Roll" },
+          { label: "User Label" },
+          { label: "Yaw" },
+        ]).and.to.have.lengthOf(18);
+
+        expect(content!.contentSet.length).to.eq(4);
+      });
+
+      it("uses `calculatedProperties` attribute", async () => {
+        // __PUBLISH_EXTRACT_START__ ContentInstancesOfSpecificClasses.calculatedProperties.Ruleset
+        // The specification returns content for `bis.SpatialViewDefinition` and a custom
+        // calculated property `Camera view direction`.
+        const ruleset: Ruleset = {
+          id: "example",
+          rules: [{
+            ruleType: RuleTypes.Content,
+            specifications: [{
+              specType: ContentSpecificationTypes.ContentInstancesOfSpecificClasses,
+              classes: { schemaName: "BisCore", classNames: ["SpatialViewDefinition"] },
+              calculatedProperties: [{
+                label: "Camera view direction",
+                value: "IIf (this.pitch >= 10, \"Vertical upwards\", IIf (this.pitch <= -10, \"Vertical downwards\", \"Horizontal\"))",
+              }],
+            }],
           }],
-        }],
-      };
-      // __PUBLISH_EXTRACT_END__
-
-      // Ensure all `bis.GeometricElement3d` properties are in the custom category
-      const content = await Presentation.presentation.getContent({
-        imodel,
-        rulesetOrId: ruleset,
-        keys: new KeySet([{ className: "BisCore:Element", id: "0x61" }]),
-        descriptor: {},
+        };
+        // __PUBLISH_EXTRACT_END__
+
+        // Ensure that derived `bis.DisplayStyle` instance properties are also returned with `bis.SpatialViewDefinition` content.
+        const content = await Presentation.presentation.getContent({
+          imodel,
+          rulesetOrId: ruleset,
+          keys: new KeySet(),
+          descriptor: {},
+        });
+        expect(content!.descriptor.fields).to.containSubset([
+          { label: "Camera view direction" },
+          { label: "Category Selector" },
+          { label: "Code" },
+          { label: "Description" },
+          { label: "Display Style" },
+          { label: "Extents" },
+          { label: "Eye Point" },
+          { label: "Focus Distance" },
+          { label: "Is Camera On" },
+          { label: "Is Private" },
+          { label: "Lens Angle" },
+          { label: "Model" },
+          { label: "Model Selector" },
+          { label: "Origin" },
+          { label: "Pitch" },
+          { label: "Roll" },
+          { label: "User Label" },
+          { label: "Yaw" },
+        ]).and.to.have.lengthOf(18);
+
+        expect(content!.contentSet.length).to.eq(4);
```

Comment:
This seems to be mostly copied from the previous `relatedProperties` test. Instead, I think we should be concentrating on the calculated property more and remove the checks for other properties.